### ISSUE TITLE
status "stopped" instead of "spawning", tick() method of LoadShape called only once

### ISSUE THREAD

**Issue by stanislawskwark** [locustio/locust#1762]
### Describe the bug
This happens with high number of workers, for me it sometimes happens when I use ~80 workers, it does not happen each time though, I didn't find the root cause so far. It never happened for me when running standalone locust, with low number of workers it happened just once as far as I know. 
Sometimes build starts with "stopped" status instead of "spawning". Number of users grows until it reaches expected value, then status changes to "running". The most problematic thing for me is that when it happens, locust stops calling tick() method of load shape, so shape doesn't really work and number of users get stuck on value returned in first call of tick() method.
### Expected behavior
Status shown in GUI represents actual status and tick method is called every second until it returns None or the build is stopped manually

### Actual behavior
GUI hows status "stopped" instead of "spawning", tick() method of LoadShape called only once

### Steps to reproduce
set up master and around 80 workers (maybe it would happen more often with higher number of workers, I didn't have access to more machines to check it)
### Environment

- OS: centos 7
- Python version: 3.8.2
- Locust version: 1.4.3, 1.4.4, 1.5.2
- Locust command line that you ran: locust --master --skip-log-setup --only-summary
- Locust file contents (anonymized if necessary): this happens regardless of locust file, even with dummy locust file which doesn't do any requests, just task which creates random list

**Comment by cyberw** [comment]
If you can, then have a look at fixing it your self @stanislawskwark . Or maybe @max-rocket-internet can have a look.

**Comment by stanislawskwark** [comment]
@cyberw I would be happy to fix it, I would need some guidance though. from what I checked until now - as I suspected shape failure is direct consequence of changes state - tick() is called only if state is STATE_INIT, STATE_SPAWNING or STATE_RUNNING.

from what I see first two calls are within milliseconds, and state is set to STOPPED, so my guess is that `check_stopped` is misbehaving on higher load and 
`        if (
            not self.state == STATE_INIT
            and not self.state == STATE_STOPPED
            and all(map(lambda x: x.state != STATE_RUNNING and x.state != STATE_SPAWNING, self.clients.all))
        )
`  evaluates to True because correct status was not set yet, could somebody confirm or give me some hint?
